File: Object-C/BlackFireIM/Example/Lib/MSIMSDK-UIKit/ChatDetail/View/More/MSInputMoreCell.m
//
//  BFInputMoreCell.m
//  BlackFireIM
//
//  Created by benny wang on 2021/3/22.
//

#import "MSInputMoreCell.h"


@implementation MSInputMoreCellData

- (instancetype)initWithType:(MSIMMoreType)type
{
    if (self = [super init]) {
        _tye = type;
    }
    return self;
}

@end

@implementation MSInputMoreCell

- (id)initWithFrame:(CGRect)frame
{
    self = [super initWithFrame:frame];
    if(self){
        [self setupViews];
    }
    return self;
}

- (void)setupViews
{
    _image = [[UIImageView alloc] init];
    _image.contentMode = UIViewContentModeScaleAspectFit;
    [self addSubview:_image];

    _title = [[UILabel alloc] init];
    [_title setFont:[UIFont systemFontOfSize:14]];
    [_title setTextColor:[UIColor grayColor]];
    _title.textAlignment = NSTextAlignmentCenter;
    [self addSubview:_title];
}

- (void)fillWithData:(MSInputMoreCellData *)data
{
    //set data
    _data = data;
    _image.image = data.image;
    [_title setText:data.title];
    //update layout
    CGSize menuSize = CGSizeMake(50, 50);
    _image.frame = CGRectMake(7.5, 7.5, menuSize.width, menuSize.height);
    _title.frame = CGRectMake(0, _image.frame.origin.y + _image.frame.size.height + 12, _image.frame.size.width + 15, 20);
}

+ (CGSize)getSize
{
    CGSize menuSize = CGSizeMake(50, 50);
    return CGSizeMake(menuSize.width, menuSize.height + 40);
}

@end

File: Swift/MSIM-Swift/Lib/MSIMSDK-UIKit/ChatDetail/View/More/MSInputMoreCell.swift
//
//  MSInputMoreCell.swift
//  MSIM-Swift
//
//  Created by benny wang on 2021/8/3.
//

import UIKit

open class MSInputMoreCell: UICollectionViewCell {
    
    var image: UIImageView!
    
    var title: UILabel!
    
    var data: MSInputMoreCellData?
    
    func fillWithData(data: MSInputMoreCellData?) {
        self.data = data
        image.image = data?.image
        title.text = data?.title
        
        let menuSize = CGSize(width: 50, height: 50)
        image.frame = CGRect(x: 7.5, y: 7.5, width: menuSize.width, height: menuSize.height)
        title.frame = CGRect(x: 0, y: image.top + image.height + 10, width: image.width + 15, height: 20)
    }
    
    static func getSize() -> CGSize {
        let menuSize = CGSize(width: 50, height: 50)
        return CGSize(width: menuSize.width, height: menuSize.height + 40)
    }
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        
        image = UIImageView()
        image.contentMode = .scaleAspectFit
        addSubview(image)
        
        title = UILabel()
        title.font = .systemFont(ofSize: 14)
        title.textColor = .gray
        title.textAlignment = .center
        addSubview(title)
    }
    
    required public init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
}

public enum MSIMMoreType {
    case photo
    case video
    case location
    case voiceCall
    case videoCall
}
public struct MSInputMoreCellData {
    
    var type: MSIMMoreType
    
    var image: UIImage?
    
    var title: String?
    
    init(type: MSIMMoreType) {
        self.type = type
    }
}
